### PR TITLE
fix:ts issue and azure eu support [ECO-3025]

### PR DESCRIPTION
**Pull request by Amitkanswal**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=eTNiFGd)
### Changes-
- Updated dependencies 
- Updated TS support
- Updated Azure EU support

### Story-[ECO-3025]

### CODE REVIEW

File: src/components/app-blog-post/app-blog-post.tsx [hunk 1]
```diff
@@ -81,7 +81,7 @@ export class AppBlogPost {
           </div>
           <div class="blog-column-right">
             <div class="related-post">
-              {banner != {} && banner.page_components && <h2 {...banner.page_components[2].widget.$?.title_h2}>{banner.page_components[2].widget.title_h2}</h2>}
+              {Object.keys(banner).length && banner.page_components && <h2 {...banner.page_components[2].widget.$?.title_h2}>{banner.page_components[2].widget.title_h2}</h2>}
```
Comment (surajcontentstack) [inline src/components/app-blog-post/app-blog-post.tsx:84]: you can use `isEmpty` from lodash

File: src/components.d.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@
  */
 import { HTMLStencilElement, JSXBase } from "@stencil/core/internal";
 import { MatchResults, RouterHistory } from "@stencil/router";
+export { MatchResults, RouterHistory } from "@stencil/router";
```
[the remark this answers is not in the document]
Comment (Amitkanswal) [reply src/components.d.ts:9]: This file is auto-created by stencil

File: src/components/blog-section.tsx [hunk 3]
```diff
@@ -17,21 +12,21 @@ export default function BlogSection(props: FeaturedBlogProps) {
   return (
     <div class="community-section">
       <div class="community-head">
-        {fromBlog.title_h2 && <h2 {...fromBlog.$?.title_h2}>{fromBlog.title_h2}</h2>}
+        {fromBlog.title_h2 && <h2 {...fromBlog.$?.title_h2 as {}}>{fromBlog.title_h2}</h2>}
```
[the remark this answers is not in the document]
Comment (Amitkanswal) [reply src/components/blog-section.tsx:15]: > ok I will update it



> i think its better to use `as object`

Not able to update it coz if we do that ts is comparing it with pre-defined properties of Object and throwing an error

File: src/components/app-header/app-header.tsx [hunk 4]
```diff
@@ -1,53 +1,43 @@
+/* eslint-disable @stencil/own-methods-must-be-private */
+/* eslint-disable @stencil/strict-boolean-conditions */
```
Comment (abhishek305) [inline src/components/app-header/app-header.tsx:2]: can we remove this ? or place it in the eslint file

**Review by abhishek305 — APPROVED** [review]
minor suggestion LGTM

File: src/components/app-header/app-header.tsx (hunk 4 above)
Comment (Amitkanswal) [reply src/components/app-header/app-header.tsx:2]: yes. I missed it

**Review by MaitriGurey1 — APPROVED** [review]

File: src/components/app-footer/app-footer.tsx [hunk 5]
```diff
@@ -16,21 +17,20 @@ export class AppFooter {
...
-  buildNavigation(ent: Entry, ft: FooterProps) {
-    let newFooter = { ...ft };
-    if (ent.length !== newFooter.navigation.link.length) {
-      ent.forEach(entry => {
-        const fFound = newFooter?.navigation.link.find(nlink => nlink.title === entry.title);
+  buildNavigation(entries: PageProps[], footerRes: FooterRes) {
+    const navFooterList = footerRes.navigation.link;
+    if (entries.length !== footerRes.navigation.link.length) {
+      entries.forEach((entry) => {
+        const fFound = footerRes.navigation.link.find(
```
Comment (surajcontentstack) [inline src/components/app-footer/app-footer.tsx:24]: suggestion: could have used `find` from lodash
`find(footerRes.navigation.link, {title: entry.title})`

**Review by surajcontentstack — APPROVED** [review]

**Pull request merged**